Change in behavior when using both pyfakefs and monkeypatch fixtures

**Describe the bug**
While upgrading pyfakefs, I noticed some of our tests failing during pytest teardown, and realized it was because pyfakefs was still active after tests ran. I tracked it down to an issue that is easy to fix in our test code (likely due to someone misunderstanding how to use pyfakefs) but wanted to report it in case others run into it or other related issues when upgrading.

**How To Reproduce**
Test file 
```python
from pyfakefs.fake_filesystem import FakeFileOpen

def test_monkeypatch_with_fs(fs, monkeypatch):
    """ monkeypatches open even though the fs fixture already does that """
    fake_open = FakeFileOpen(fs)
    monkeypatch.setattr("builtins.open", fake_open, raising=False)

def test_open():
    """ Tests if open is poisoned by the above test"""
    assert "built-in" in str(open)
```

My guess is that, in the above test, pyfakefs patches builtins.open, and then monkeypatch patches it again. I believe monkeypatch stores the previous value when patching, so when the test is torn down, it can revert the patch. If I'm right, monkeypatch will revert it to the value that pyfakefs had already replaced it with, but I don't know why pyfakefs wouldnt then revert it again to the original builtins.open.

I ran the tests using uv like this:

```shell
$ uvx -p 3.12  --with pyfakefs==5.3.5 --with pytest pytest -v foo.py
Installed 6 packages in 11ms
Cannot read termcap database;
using dumb terminal settings.
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.12.0, pytest-8.4.1, pluggy-1.6.0 -- /home/jceresini/.cache/uv/archive-v0/beQ-4Zy-5L-JDvHTVdL6W/bin/python
cachedir: .pytest_cache
rootdir: /home/jceresini/repos/cloudn2/.wt/upgrade_pyfakefs
configfile: pyproject.toml
plugins: pyfakefs-5.3.5
collected 2 items                                                                                                                                                                                                                            

foo.py::test_monkeypatch_with_fs PASSED                                                                                                                                                                                                [ 50%]
foo.py::test_open PASSED                                                                                                                                                                                                               [100%]

============================================================================================================= 2 passed in 0.02s ==============================================================================================================


$ uvx -p 3.12  --with pyfakefs==5.9.1 --with pytest pytest -v foo.py
Installed 6 packages in 6ms
Cannot read termcap database;
using dumb terminal settings.
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.12.0, pytest-8.4.1, pluggy-1.6.0 -- /home/jceresini/.cache/uv/archive-v0/GtnfJ2GjL-vg9q6DKgFjd/bin/python
cachedir: .pytest_cache
rootdir: /home/jceresini/repos/cloudn2/.wt/upgrade_pyfakefs
configfile: pyproject.toml
plugins: pyfakefs-5.9.1
collected 2 items                                                                                                                                                                                                                            

foo.py::test_monkeypatch_with_fs PASSED                                                                                                                                                                                                [ 50%]
foo.py::test_open FAILED                                                                                                                                                                                                               [100%]

================================================================================================================== FAILURES ==================================================================================================================
_________________________________________________________________________________________________________________ test_open __________________________________________________________________________________________________________________

    def test_open():
        """ Tests if open is poisoned by the above test"""
>       assert "built-in" in str(open)
E       AssertionError: assert 'built-in' in '<bound method FakeIoModule.open of <pyfakefs.fake_io.FakeIoModule2 object at 0x7f38c21291c0>>'
E        +  where '<bound method FakeIoModule.open of <pyfakefs.fake_io.FakeIoModule2 object at 0x7f38c21291c0>>' = str(open)

foo.py:10: AssertionError
========================================================================================================== short test summary info ===========================================================================================================
FAILED foo.py::test_open - AssertionError: assert 'built-in' in '<bound method FakeIoModule.open of <pyfakefs.fake_io.FakeIoModule2 object at 0x7f38c21291c0>>'
======================================================================================================== 1 failed, 1 passed in 0.05s =========================================================================================================

```

It was an easy fix in my case, the monkeypatch was completely unnecessary. Reporting in case there's some issue with pytest setup/teardown that may cause other impact.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -15,6 +15,9 @@ The released versions correspond to PyPI releases.
 
 ## Unreleased
 
+### Fixes
+* fixed an interaction problem of `fs` with other pytest fixtures (see [#1200](../../issues/1200))
+
 ### Infrastructure
 * fixed some warnings in tests (see [#1190](../../issues/1190))
 

--- a/docs/usage.rst
+++ b/docs/usage.rst
@@ -43,6 +43,9 @@ you can define a longer name in your ``conftest.py`` and use that in your tests:
         """
         yield fs
 
+.. note:: Filesystem patching is paused in the pytest logreport phases to ensure that
+  logs are correctly written to the real filesystem.
+
 Class-, module- and session-scoped fixtures
 ...........................................
 For convenience, class-, module- and session-scoped fixtures with the same
@@ -54,6 +57,9 @@ respectively.
   will just serve as a reference to the active fake filesystem. That means that changes
   done in the fake filesystem inside a test will remain there until the respective scope
   ends (see also :ref:`nested_patcher_invocation`).
+
+.. note:: To avoid unwanted side-effects, the patching is paused between the tests,
+  even if the fixture is still active.
 
 .. _unittest_usage:
 

--- a/pyfakefs/fake_filesystem_unittest.py
+++ b/pyfakefs/fake_filesystem_unittest.py
@@ -789,7 +789,7 @@ class Patcher:
             "io": fake_io.FakeIoModule,
             "pathlib": fake_pathlib_module,
         }
-        if sys.version_info >= (3, 13):
+        if sys.version_info[:2] == (3, 13):
             # for Python 3.13, we need both pathlib (path with __init__.py) and
             # pathlib._local (has the actual implementation);
             # depending on how pathlib is imported, either may be used
@@ -806,10 +806,10 @@ class Patcher:
         # be contained in - this allows for alternative modules like
         # `pathlib` and `pathlib2`
         self._class_modules["Path"] = ["pathlib"]
-        if sys.version_info >= (3, 13):
+        if sys.version_info[:2] == (3, 13):
             self._class_modules["Path"].append("pathlib._local")
         self._unfaked_module_classes["pathlib"] = fake_pathlib.RealPathlibModule
-        if sys.version_info >= (3, 13):
+        if sys.version_info[:2] == (3, 13):
             self._unfaked_module_classes["pathlib._local"] = (
                 fake_pathlib.RealPathlibModule
             )

--- a/pyfakefs/pytest_plugin.py
+++ b/pyfakefs/pytest_plugin.py
@@ -9,11 +9,13 @@ def my_fakefs_test(fs):
     assert os.path.exists('/var/data/xx1.txt')
 """
 
+import contextlib
+
 import py
 import pytest
 from _pytest import capture
 
-from pyfakefs.fake_filesystem_unittest import Patcher
+from pyfakefs.fake_filesystem_unittest import Patcher, Pause
 
 try:
     from _pytest import pathlib
@@ -85,22 +87,18 @@ def pytest_sessionfinish(session, exitstatus):
 @pytest.hookimpl(hookwrapper=True, tryfirst=True)
 def pytest_runtest_logreport(report):
     """Make sure that patching is not active during reporting."""
-    pause = Patcher.PATCHER is not None and report.when == "call"
-    if pause:
+    pause = Patcher.PATCHER is not None and Patcher.PATCHER.is_patching
+    context_mgr = Pause(Patcher.PATCHER) if pause else contextlib.nullcontext()
+    with context_mgr:
+        yield
+    if pause and report.when == "teardown":
+        # if we get here, we are not in a function scope fixture
+        # in this case, we still want to pause patching between the tests
         Patcher.PATCHER.pause()
-    yield
 
 
-@pytest.hookimpl(hookwrapper=True, trylast=True)
+@pytest.hookimpl(tryfirst=True)
 def pytest_runtest_setup(item):
+    # resume patcher if not in a function scope
     if Patcher.PATCHER is not None:
         Patcher.PATCHER.resume()
-    yield
-
-
-@pytest.hookimpl(hookwrapper=True, tryfirst=True)
-def pytest_runtest_teardown(item, nextitem):
-    """Make sure that patching is not active during reporting."""
-    if Patcher.PATCHER is not None:
-        Patcher.PATCHER.pause()
-    yield

--- a/pyfakefs/pytest_tests/test_fs_with_monkeypatch.py
+++ b/pyfakefs/pytest_tests/test_fs_with_monkeypatch.py
@@ -1,0 +1,12 @@
+from pyfakefs.fake_filesystem import FakeFileOpen
+
+
+def test_monkeypatch_with_fs(fs, monkeypatch):
+    """Regression test for issue 1200"""
+    fake_open = FakeFileOpen(fs)
+    monkeypatch.setattr("builtins.open", fake_open, raising=False)
+
+
+def test_open():
+    """Tests if open is poisoned by the above test"""
+    assert "built-in" in str(open)

